lerobot/common/policies/vqbet/modeling_vqbet.py

```diff
@@ -0,0 +1,2460 @@
+            + list(policy.vqbet.action_head.vqvae_model.decoder.parameters())
+            + list(policy.vqbet.action_head.vqvae_model.vq_layer.parameters())
+        )
+        self.vqvae_optimizer = torch.optim.Adam(
+            vqvae_params, lr=cfg.training.vqvae_lr, weight_decay=0.0001
+        )
+
+        self.encoder_optimizer = torch.optim.Adam(
+            policy.vqbet.rgb_encoder.parameters(),
+            cfg.training.lr,
+            cfg.training.adam_betas,
+            cfg.training.adam_eps,
+            cfg.training.adam_weight_decay,
+        )
+
+        self.bet_optimizer1 = policy.vqbet.policy.configure_optimizers(
+            weight_decay=cfg.training.bet_weight_decay,
+            learning_rate=cfg.training.bet_learning_rate,
+            betas=cfg.training.bet_betas,
+        )
+
+        self.bet_optimizer1.add_param_group(
+                {"params": policy.vqbet._action_token}
+            )
+        self.bet_optimizer1.add_param_group(
+                {"params": policy.vqbet.state_projector.parameters()}
+            )
+        self.bet_optimizer1.add_param_group(
+                {"params": policy.vqbet.rgb_feature_projector.parameters()}
+            )
+
+        if cfg.policy.sequentially_select:
+            vqbet_head_params = (
+                list(policy.vqbet.action_head.map_to_cbet_preds_primary_bin.parameters())
+                + list(policy.vqbet.action_head.map_to_cbet_preds_secondary_bin.parameters())
+                + list(policy.vqbet.action_head.map_to_cbet_preds_offset.parameters())
+            )
+
+        else:
+            vqbet_head_params = (
+                list(policy.vqbet.action_head.map_to_cbet_preds_bin.parameters())
+                + list(policy.vqbet.action_head.map_to_cbet_preds_offset.parameters())
+            )
+        self.bet_optimizer2 = torch.optim.AdamW(
+            vqbet_head_params,
+            lr=cfg.training.bet_learning_rate,
+            weight_decay=cfg.training.bet_weight_decay,
+            betas=cfg.training.bet_betas,
+        )
+
+        self.param_groups = self.encoder_optimizer.param_groups
+
+    def step(self):
+        self.optimizing_step +=1
+        # pretraining VQ-VAE (Training Phase 1)
+        if self.optimizing_step < self.n_vqvae_training_steps:
+            self.vqvae_optimizer.step()
+        # training BeT (Training Phase 2)
+        else:
+            self.encoder_optimizer.step()
+            self.bet_optimizer1.step()
+            self.bet_optimizer2.step()
+
+    def zero_grad(self):
+        # pretraining VQ-VAE (Training Phase 1)
+        if self.optimizing_step < self.n_vqvae_training_steps:
+            self.vqvae_optimizer.zero_grad()
+        # training BeT (Training Phase 2)
+        else:
+            self.encoder_optimizer.zero_grad()
+            self.bet_optimizer1.zero_grad()
+            self.bet_optimizer2.zero_grad()
+
+class VQBeTScheduler(nn.Module):
```

Comment:
Reflected. I put all the parameters for phase 2 in the scheduler. However, to make sure that the scheduler don't changes the `lr` during phase 1, we couldn't delete this part at all. If you have better idea for this function, please let me know:)
```
class VQBeTScheduler(nn.Module):
    def __init__(self, optimizer, cfg):
        super().__init__()
        # VQ-BeT use scheduler only for rgb encoder. Since we took rgb encoder part from diffusion policy, we also follow the same scheduler from it.
        from diffusers.optimization import get_scheduler
        self.n_vqvae_training_steps = cfg.training.n_vqvae_training_steps
        self.optimizing_step = 0

        self.lr_scheduler = get_scheduler(
            cfg.training.lr_scheduler,
            optimizer=optimizer,
            num_warmup_steps=cfg.training.lr_warmup_steps,
            num_training_steps=cfg.training.offline_steps,
        )


    def step(self):
        self.optimizing_step +=1
        if self.optimizing_step >= self.n_vqvae_training_steps:
            self.lr_scheduler.step()
```